### PR TITLE
provider added

### PR DESCRIPTION
**Pull request by Buhana149**

### CODE REVIEW

File: lib/calculator_page.dart [hunk 1]
```diff
@@ -10,147 +11,94 @@ class CalculatorPage extends StatefulWidget {
...
-                  ),
-                  Container(
-                    padding: const EdgeInsets.all(20),
-                    alignment: Alignment.centerLeft,
-                    child: Text(
-                      userQuestion,
-                      style: const TextStyle(fontSize: 20),
+     CalculatorProvider calculate =
+      Provider.of<CalculatorProvider>(context);
+    return Consumer<CalculatorProvider>(builder: (context, value, child) {
```
Comment (vlad-buhaescu) [inline lib/calculator_page.dart:16]: you can use value vairable instead of calculate variable on line 18

File: lib/calculator_page.dart [hunk 2]
```diff
@@ -10,147 +11,94 @@ class CalculatorPage extends StatefulWidget {
...
+              flex: 2,
+              child: Container(
+                child: GridView.builder(
+                  itemCount: calculate.buttons.length,
+                  gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
+                    crossAxisCount: 4,
+                  ),
+                  itemBuilder: (BuildContext context, int index) {
+                    if (index == 0) {
+                      return MyButton(
```
Comment (vlad-buhaescu) [inline lib/calculator_page.dart]: MyButton can be named more meaningful

File: lib/calculator_page.dart [hunk 3]
```diff
@@ -10,147 +11,94 @@ class CalculatorPage extends StatefulWidget {
...
+            Expanded(
+              flex: 2,
+              child: Container(
+                child: GridView.builder(
+                  itemCount: calculate.buttons.length,
+                  gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
+                    crossAxisCount: 4,
+                  ),
+                  itemBuilder: (BuildContext context, int index) {
+                    if (index == 0) {
```
Comment (vlad-buhaescu) [inline lib/calculator_page.dart]: for this try to use a switch

File: lib/calculator_page.dart [hunk 4]
```diff
@@ -10,147 +11,94 @@ class CalculatorPage extends StatefulWidget {
...
+                        buttonText: calculate.buttons[index],
+                      );
+                    } else if (index == 1) {
+                      return MyButton(
+                        buttonTapped: () => calculate.deleteButton(),
+                        color: Colors.red,
+                        textColor: Colors.white,
+                        buttonText: calculate.buttons[index],
+                      );
+                    } else if (index == calculate.buttons.length - 1) {
```
Comment (vlad-buhaescu) [inline lib/calculator_page.dart]: for switch the case will be 

case var lastIndex when index == calculate.buttons.length - 1:

File: lib/services/calculator_services.dart [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import 'package:flutter/material.dart';
+
+class CalculatorServices  {
```
Comment (vlad-buhaescu) [inline lib/services/calculator_services.dart]: this class seems unused, please remove

**Review by vlad-buhaescu — CHANGES_REQUESTED** [review]

File: lib/calculator_page.dart (hunk 1 above)
Comment (Buhana149) [reply lib/calculator_page.dart:16]: done

File: lib/calculator_page.dart (hunk 2 above)
Comment (Buhana149) [reply lib/calculator_page.dart]: done

File: lib/services/calculator_services.dart (hunk 5 above)
Comment (Buhana149) [reply lib/services/calculator_services.dart]: done

File: lib/calculator_page.dart (hunk 4 above)
Comment (Buhana149) [reply lib/calculator_page.dart]: done

File: lib/calculator_page.dart (hunk 3 above)
Comment (Buhana149) [reply lib/calculator_page.dart]: done

**Review by vlad-buhaescu — APPROVED** [review]

**Pull request merged**